### PR TITLE
Add test coverage for SAT-24197

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
### Problem Statement
Missing  test coverage for SAT-24197, to verify foreman-discovery-image is built on latest up-to-date RHEL

### Solution
Add test coverage for SAT-24197, to verify foreman-discovery-image is built on latest up-to-date RHEL

### CODE REVIEW

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_discoveredhost.py -k test_positive_verify_updated_fdi_image

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7546
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_discoveredhost.py -k test_positive_verify_updated_fdi_image --external-logging
Test Result : ========== 1 failed, 17 deselected, 26 warnings in 781.03s (0:13:01) ===========
```

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_discoveredhost.py -k test_positive_verify_updated_fdi_image

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_discoveredhost.py -k test_positive_verify_updated_fdi_image

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7574
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_discoveredhost.py -k test_positive_verify_updated_fdi_image --external-logging
Test Result : ========== 1 passed, 17 deselected, 26 warnings in 788.59s (0:13:08) ===========
```

File: tests/foreman/cli/test_discoveredhost.py [hunk 1]
```diff
@@ -412,3 +414,35 @@ def test_positive_list_facts():
...
+    :CaseImportance: Critical
+    """
+    discovery_ks_path = '/usr/share/foreman-discovery-image/foreman-discovery-image.ks'
+    target_sat.register_to_cdn()
+    target_sat.execute('yum -y --disableplugin=foreman-protector install foreman-discovery-image')
+
+    if target_sat.os_version.major == 9:
+        version = '8.9' if is_open('SAT-25275') else str(target_sat.os_version)
+    elif target_sat.os_version.major == 8:
+        version = '8.9' if is_open('SAT-24197') else str(target_sat.os_version)
```
Comment (pondrejk) [inline tests/foreman/cli/test_discoveredhost.py:445]: Not sure if it makes sense to verify that the incorrect version is present, shouldn't we query for the desired state and use blockedBy to get the test run once the bugs are closed?

**Review by pondrejk — CHANGES_REQUESTED** [review]

File: tests/foreman/cli/test_discoveredhost.py (hunk 1 above)
Comment (Gauravtalreja1) [reply tests/foreman/cli/test_discoveredhost.py:445]: I considered this way to verify as there were two different issues for EL8 and EL9, where EL9 isn't expected to be fixed any time soon and EL8 issue is already in-progress but not sure when it'll be ready as well.
And incorrect version is not blocker here it works for both EL8 and EL9 right now, we're just adding this test to verify we build FDI always on latest RHEL, so I wouldn't prefer using blockedBy for this test

**Review by ColeHiggins2 — APPROVED** [review]

File: tests/foreman/cli/test_discoveredhost.py (hunk 1 above)
Comment (pondrejk) [reply tests/foreman/cli/test_discoveredhost.py:445]: I don't want to block this, but wouldn't this implementation create confusing messages? iiuc the Verifies tag will cause these jiras to get comments that the issue is verified by automation when this test passes, even before they are fixed :) @jameerpathan111 is that right?

Comment (jameerpathan111) [reply tests/foreman/cli/test_discoveredhost.py:445]: @pondrejk No, unless the issues mentioned by`Verifies` are in "Review" or "Release Pending" state, the comment for test results won't be added. In other words, by default, it only adds comment if the issue is fixed and is on_qa and not released yet.

**Review by jameerpathan111 — APPROVED** [review]

File: tests/foreman/cli/test_discoveredhost.py (hunk 1 above)
Comment (pondrejk) [reply tests/foreman/cli/test_discoveredhost.py:445]: well, fair enough

**Review by pondrejk — APPROVED** [review]

**Pull request merged**